openupgradelib/openupgrade_merge_records.py

```diff
@@ -202,21 +202,35 @@ def _change_translations_orm(env, model_name, record_ids, target_record_id,
                              exclude_columns):
     if ('ir_translation', 'res_id') in exclude_columns:
         return
-    records = env['ir.translation'].search([
+    translation_obj = env['ir.translation']
+    groups = translation_obj.read_group([
         ('type', '=', 'model'),
         ('res_id', 'in', record_ids),
-        ('name', 'like', '%s,%%' % model_name)])
-    if records:
-        records.write({'res_id': target_record_id})
-        logger.debug(
-            "Changed %s translations of model 'ir.translation'",
-            len(records))
+        ('name', 'like', '%s,%%' % model_name),
+    ], ['name', 'lang'], ['name', 'lang'], lazy=False)
+    for group in groups:
+        target_translation = translation_obj.search([
+            ('type', '=', 'model'),
+            ('res_id', '=', target_record_id),
+            ('name', '=', group['name']),
+            ('lang', '=', group['lang']),
+        ])
```

Comment:
limit=1?

Reply:
As we are not reading values after this search, it's the same limiting or not. And in v12, this will be directly limited by the constraint, and for the rest, it's weird to have more than 1 values. We can leave it as is.

Reply:
Put perfomance-wise it's better reading one than more than one or not? :S

Reply:
Put perfomance-wise it's better reading one than more than one or not? :S

Reply:
As said, we are not reading anything, so this is going to have the same performance. Even more, limiting will add an extra operation on PG side.

Reply:
just nitpicking

Reply:
nitpicking is when there's a valid comment, but not critical. This is not the case, as you can check reading my first comment, so don't persist in this case.